structuralcodes/geometry/_reinforcement.py

```diff
@@ -125,3 +122,127 @@ def add_reinforcement_line(
             group_label=group_label,
         )
     return geo
+
+
+def add_reinforcement_circle(
+    geo: t.Union[SurfaceGeometry, CompoundGeometry],
+    center: t.Tuple[float, float],
+    radius: float,
+    diameter: float,
+    material: t.Union[Material, ConstitutiveLaw],
+    n: int = 0,
+    s: float = 0.0,
+    first: bool = True,
+    last: bool = True,
+    start_angle: float = 0.0,
+    stop_angle: float = 2 * np.pi,
+    group_label: t.Optional[str] = None,
+) -> CompoundGeometry:
+    """Adds a set of bars distributed in a line.
+
+    Arguments:
+        geo (Union(SurfaceGeometry, CompoundGeometry)): The geometry used as
+            input.
+        center (Tuple(float, float)): Coordinates of the center point of
+            the circle line where reinforcement will be added.
+        radius (float): Radius of the circle line where reinforcement will be
+            added.
+        diameter (float): The diameter of the bars.
+        material (Union(Material, ConstitutiveLaw)): A valid material or
+            constitutive law.
+        n (int): The number of bars to be distributed inside the line (default
+            = 0).
+        s (float): The distance between the bars (default = 0).
+        first (bool): Boolean indicating if placing the first bar (default =
+            True).
+        last (bool): Boolean indicating if placing the last bar (default =
+            True).
+        start_angle (float): Start angle (respect to X axis) for defining the
+            arch where to add bars in radians (default = 0)
+        stop_angle (float): Stop angle (respect to X axis) for defining the
+            arch where to add bars in radians (default = 2pi)
+        group_label (Optional(str)): A label for grouping several objects
+            (default is None).
+
+    Note:
+        At least n or s should be greater than zero.
+        Attribues start_angle and stop_angle by default are 0 and 2pi
+        respectively, so that bars will be distributed along the whole circle.
+        If only a portion of the circle must be used (i.e. an arch of
+        circumference), then set start and stop angles correspondingly.
+        stop_angle must always be larger than start_angle.
+
+    Returns:
+        CompoundGeometry: A compound geometry with the original geometry and
+        the reinforcement.
+    """
+    from math import floor
+
+    # Check that difference between stop and start angle is
+    # positive and less than 2pi
+    if stop_angle - start_angle <= 0 or stop_angle - start_angle > 2 * np.pi:
+        raise ValueError(
+            'Stop angle should be larger than start angle and difference \
+            them should be at most 2pi.'
+        )
+    # Calculate length from start_angle to stop_angle
+    length = radius * (stop_angle - start_angle)
+
+    # If the whole circle, than deal with the case that would add an extra bar
+    whole = False
+    add_n = 0
+    if abs(length - 2 * np.pi * radius) < 1e-4:
```

Comment:
Does it make sense to instead use `math.isclose` here?

Reply:
I am not sure I am following this suggestion...

Reply:
Sure! done in 87064519bacb2ab2efbe9ef68fb8e8426df07e90